gem-push-action

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/bash 
+set -e -o pipefail
+
+if parse-gemspec --is-pre-release; then
+    if [[ $INPUT_PRE_RELEASE != true ]]; then
+    echo Ignoring pre-release. Set input pre-release: true, to release
+    exit 0
+    fi
+elif [[ $INPUT_RELEASE != true ]]; then
+    echo Ignoring release. Set input release: true, to release
+    exit 0
+fi
+
+# tee the output to get it in the logs but capture as we can't tell why gem
+# push failed from the exit code, so need to grep the output. Gem existing is
+# ok, other errors not. Avoids playing games setting up auth differently for
+# gem query.
+# Note: the glob is intentially unquoted, we want a glob!
+if ! gem push --host "$GEM_HOST" $INPUT_PACKAGE_GLOB | tee push.out; then
```

Comment:
I still don't understand why to use `| tee push.out` instead of `> push.out`. Why print errors twice in the log?

Reply:
If the error is not something other than existing gem (e.g. bad creds) and we do `if gem push | grep 'already pushed'` then the error message will be lost.